app/controllers/ems_common.rb

```diff
@@ -427,27 +427,66 @@ def update_button_reset
 
   def update_button_validate
     verify_ems = find_by_id_filtered(model, params[:id])
-    validate_credentials verify_ems
+    queue_credential_validation_task verify_ems
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def queue_credential_validation_task(verify_ems)
     set_record_vars(verify_ems, :validate)
+    creds = set_record_creds(verify_ems)
+
     @in_a_form = true
     @changed = session[:changed]
 
     # validate button should say "revalidate" if the form is unchanged
     revalidating = !edit_changed?
-    result, details = verify_ems.authentication_check(params[:type], :save => revalidating)
-    if result
-      add_flash(_("Credential validation was successful"))
+
+    args = {
+      :ems        => verify_ems,
+      :creds      => creds,
+      :auth_type  => params[:type],
```

Comment:
It may be a security issue, but it's an existing security issue (see [deleted line 441](https://github.com/ManageIQ/manageiq/pull/4869#discussion-diff-42268053L441)).

In general, `ems_common` needs to be refactored to pull out all the non-common stuff.  And, I think the `params[:type]` will be removed in that effort.  But, I don't think trying to find a new way to handle `param[:type]` is a job for this PR.

@Fryguy, thoughts?